src/common/collection_column_specs/load_specs.py

```diff
@@ -26,8 +47,29 @@ def load_spec(data_product: str, collection: str = None) -> ColumnarAttributesSp
         omitted, all specs for that collection are merged together, throwing an error if keys
         conflict.
     """
-    if not collection:
-        raise ValueError("I ain't done this yet, dang")
-    # TODO FILTER load multiple specs and merge them, throwing errors on keys with different specs.
-    with open(SPEC_DIR / f"{data_product}-{collection}{YML_SUFFIX}") as f:
-        return ColumnarAttributesSpec(**yaml.safe_load(f))
+    if collection:
+        coll2file = {collection: _filename(data_product, collection)}
+    else:
+        coll2file = {_specfile_to_collection(f): SPEC_DIR / f
+                     for f in _get_spec_files(data_product)}
+    coll2spec = {}
+    for coll, specfile in coll2file.items():
+        with open(specfile) as f:
+            coll2spec[coll] = {d["key"]: d for d in yaml.safe_load(f)["columns"]}
+    keys = set()
+    for v in coll2spec.values():
+        keys |= v.keys()
+    columns = []
+    for k in keys:
+        first = None
+        for coll, spec in coll2spec.items():
+            if k in spec:
+                if not first:
+                    first = (coll, spec[k])
+                else:
+                    if spec[k] != first[1]:
+                        colls = sorted([coll, first[0]])
```

Comment:
Not quite sure why you are sorting here. Shouldn't raise error directly?

Reply:
If I don't sort, then which coll name goes where in the error message is indeterminate, and the test passing is therefore indeterminate. The lack of a sort is why the test were passing locally but not in GHA